lambda-functions/jira-webhook-receiver/index.js

```diff
@@ -1,227 +1,263 @@
-'use strict';
-
-const moment = require('moment');
 const JiraConnector = require('jira-connector');
 const config = require('./config');
 const slackUtils = require('./slack-utils');
-const WebClient = require('@slack/client').WebClient;
-const web = new WebClient(config.slack.api_token);
-
-const stakeholders_field = 'customfield_10700';
-const groups_watch_field = 'customfield_11000';
-const business_verticals_field = 'customfield_12200';
+const { WebClient } = require('@slack/client');
 
-const getStakeHolders = (issue) => {
-  const stakeHolderObjs = issue.fields[stakeholders_field] || [];
-  const stakeHolders = stakeHolderObjs.map(function(obj) {
-    return obj.value;
-  }) || [];
-  return stakeHolders;
-}
+const web = new WebClient(config.slack.api_token);
 
-const getBusinessVerticals = (issue) => {
-  const businessVerticalObjs = issue.fields[business_verticals_field] || [];
-  const businessVerticals = businessVerticalObjs.map(function(obj) {
-    return obj.value;
-  }) || [];
-  return businessVerticals;
-}
+const getValueOrNameFromArray = (array) => {
+  let values = [];
 
-const groupsThatShouldFollowIssue = (issue) => {
-  if (!issue) {
-    return [];
+  if (Array.isArray(array)) {
+    values = array.map(element => element.value || element.name);
   }
 
-  const stakeHolders = getStakeHolders(issue);
-  const businessVerticals = getBusinessVerticals(issue);
+  return values;
+};
 
-  const groups = [];
+const testMatch = (valueTest, value, ...extras) => {
```

Comment:
i would rename this from valueTest to rule/condition

Reply:
checkRule(rule,value, ...extras) may be

Reply:
+1